src/linear_algebra/pi_tensor_product.lean

```diff
@@ -403,25 +360,10 @@ open_locale tensor_product
 
 variables {ι : Type*} {R : Type*} [comm_ring R]
 variables {s : ι → Type*} [∀ i, add_comm_group (s i)] [∀ i, module R (s i)]
-variables {E : Type*} [add_comm_group E] [semimodule R E]
 
 /- Unlike for the binary tensor product, we require `R` to be a `comm_ring` here, otherwise
 this is false in the case where `ι` is empty. -/
 instance : add_comm_group (⨂[R] i, s i) := semimodule.add_comm_monoid_to_add_comm_group R
 
-lemma neg_tprod_coeff (z : R) (f : Π i, s i) : tprod_coeff R (-z) f = -tprod_coeff R z f :=
-by rw [←neg_one_smul R, ←smul_tprod_coeff', neg_one_smul R]
-
-lemma neg_tprod (f : Π i, s i) (i : ι) : tprod R (update f i (-(f i))) = -tprod R f :=
-by rw [←neg_one_smul R, ←smul_tprod, neg_one_smul R]
-
-lemma neg_tprod' (f : Π i, s i) (i : ι) (m : s i) :
-  tprod R (update f i (-m)) = -tprod R (update f i m) :=
-(mk R s).map_neg f i m
```

Comment:
The former is a less general version of the latter, the latter is `(tprod R).map_neg` as you already know